### ISSUE TITLE
[FEATURE_REQUEST] Dynamic Music

### ISSUE THREAD

**Issue by deffcolony** [SillyTavern/SillyTavern#501]
**Have you searched for similar [requests](https://github.com/Cohee1207/SillyTavern/issues?q=)?**
Yes

## Is your feature request related to a problem? Please describe.
Yes, the current chat system lacks a dynamic music feature that allows users to personalize their experience by uploading their own music files or using YouTube links for background music. Additionally, the system lacks the ability to automatically switch music based on the conversation's context, scene, or mood.

## Describe the solution you'd like
I would like to request the implementation of a dynamic music feature in SillyTavern. This feature will empower users to enhance their chat experience by incorporating personalized music and enabling automated music switching based on the conversation's context.

The dynamic music feature should include the following functionalities:

### User Uploads: 
Users can upload their own music files in formats such as MP3, WAV, and OGG. These files will be stored in the designated folder: **\SillyTavern\public\sounds\music**. Users can easily upload or drop their music files into this folder.

### YouTube Link Support: 
Users can also provide YouTube links that will play the audio without video. SillyTavern will have a control panel where users can manage and control their music playlist.

### Automated Music Switching: 
The AI algorithm will scan the conversation and analyze the background, mood, and context to determine the appropriate music. Based on the conversation's content and cues, the AI will select the most suitable music from the available playlist.

### Tagging System: 
Users can add tags to their uploaded music files. These tags will help users filter and organize their playlist, making it easier to quickly find and play their favorite music styles.

### Playlist Management:
A user-friendly interface for managing the music playlist. Users will have control over their music playback experience by being able to add new tracks, remove existing tracks, reorder the playlist, and create custom playlists. This playlist management feature will provide users with a high level of control and customization, allowing them to curate their music selection according to their preferences.

## Describe alternatives you've considered
As an alternative, users can manually switch music during the chat session by manually selecting tracks from a pre-defined list (playlist). However, this process can be time-consuming and interrupts the flow of conversation. By introducing dynamic music with automated switching, users can enjoy a seamless and immersive experience without the need for constant manual intervention.

## Additional context
The dynamic music feature will significantly enhance the chat experience by allowing users to personalize their background music and create a more immersive environment. The AI algorithm's ability to analyze the conversation and select appropriate music based on context, scene, and mood will further enhance the storytelling and role-playing experience within SillyTavern. By incorporating user-uploaded music and automated music switching, SillyTavern will provide a rich and engaging platform for users to enjoy interactive chats with enhanced audio experiences.

**Comment by IkariDevGIT** [comment]
@RossAscends why not use this? https://huggingface.co/joeddav/distilbert-base-uncased-go-emotions-student
we could make like a slider(called frequency) and do some math magic like: if the current label "fear" is 0.02... not more or less then before do not switch the music, if its over 0.02 switch the music file(that is bound to the specific label) to the highest rated label. its super primitive, but it IS achivable!

**Comment by IkariDevGIT** [comment]
Hey, i would really like that feature, it would actually not be super difficult to make it..

**Comment by Woisek** [comment]
I wonder how much people use 'dynamic music' when they read a book ... 🤔

**Comment by IkariDevGIT** [comment]
i dont know, this doesent apply here tho as this is not a book 😄 its roleplay

**Comment by Cohee1207** [comment]
We might have this developed soon as an external extension. No ETAs and guarantees, however.

**Comment by Woisek** [comment]
> i dont know, this doesent apply here tho as this is not a book 😄 its roleplay

Right. And who would listen to 'dynamic music' when reading and writing half a book ... ? 😉

**Comment by IkariDevGIT** [comment]
> We might have this developed soon as an external extension. No ETAs and guarantees, however.

thanks for the answer, if its not too bothering to ask, how are you going to check which music needs to be played?

**Comment by Tony-sama** [comment]
May be simple to do the mood detection on extras server side using the existing classification extension. The UI part is the biggest task I think.

**Comment by Tony-sama** [comment]
Also, their would be no additional cost in computation if it is injected just after retrieving classify response.
